tests/serve_request.rs

```diff
@@ -130,79 +101,47 @@ fn serves_requests_hidden_files(tmpdir: TempDir, port: u16) -> Result<(), Error>
         }
     }
 
-    child.kill()?;
-
     Ok(())
 }
 
 #[rstest]
-fn serves_requests_no_hidden_files_without_flag(tmpdir: TempDir, port: u16) -> Result<(), Error> {
-    let mut child = Command::cargo_bin("miniserve")?
-        .arg(tmpdir.path())
-        .arg("-p")
-        .arg(port.to_string())
-        .stdout(Stdio::null())
-        .spawn()?;
-
-    sleep(Duration::from_secs(1));
-
-    let body = reqwest::blocking::get(format!("http://localhost:{}", port).as_str())?
-        .error_for_status()?;
+fn serves_requests_no_hidden_files_without_flag(server: TestServer) -> Result<(), Error> {
+    let body = reqwest::blocking::get(server.url())?.error_for_status()?;
     let parsed = Document::from_read(body)?;
 
     for &hidden_item in HIDDEN_FILES.into_iter().chain(HIDDEN_DIRECTORIES) {
         assert!(parsed
             .find(|x: &Node| x.text() == hidden_item)
             .next()
             .is_none());
-        let resp =
-            reqwest::blocking::get(format!("http://localhost:{}/{}", port, hidden_item).as_str())?;
+        let resp = reqwest::blocking::get(server.url().join(&hidden_item)?)?;
         assert_eq!(resp.status(), StatusCode::BAD_REQUEST);
     }
 
-    child.kill()?;
-
     Ok(())
 }
 
 #[rstest(no_symlinks, case(true), case(false))]
-fn serves_requests_symlinks(tmpdir: TempDir, port: u16, no_symlinks: bool) -> Result<(), Error> {
-    let mut comm = Command::cargo_bin("miniserve")?;
-    comm.arg(tmpdir.path())
-        .arg("-p")
-        .arg(port.to_string())
-        .stdout(Stdio::null());
-    if no_symlinks {
-        comm.arg("--no-symlinks");
-    }
-
-    let mut child = comm.spawn()?;
-    sleep(Duration::from_secs(1));
-
+fn serves_requests_symlinks(no_symlinks: bool) -> Result<(), Error> {
+    let server = server(["--no-symlinks"].iter().filter(|_| no_symlinks));
```

Comment:
I think this is easier to read:
```
let server = server(if no_symlinks {
    &["--no-symlinks"]
} else {
    &[] as &[&str]
});
```

Reply:
I'd prefer it as provided by @Jikstra in this example.